pkg/controller/add_cluster_test.go

```diff
@@ -55,6 +61,35 @@ func registerValidConfigMap() error {
 	return nil
 }
 
+func TestEnsurePlacement(t *testing.T) {
+	cluster := getFixture("cluster-3-zones.yaml", t)
+	deps := newTestDeps(t, &testOpts{
+		crdObjects: []runtime.Object{cluster},
+	})
+	k8sops, err := newFakeK8sops()
+	require.NoError(t, err)
+
+	placementMock := deps.placementClient
+	defer deps.cleanup()
+
+	controller := deps.newController()
+	controller.k8sclient = k8sops
+
+	placementMock.EXPECT().Get().Return(nil, pkgerrors.WithMessage(m3admin.ErrNotFound, "foo"))
+	placementMock.EXPECT().Init(gomock.Any())
+
+	err = controller.EnsurePlacement(cluster)
+	assert.NoError(t, err)
+
+	placementMock.EXPECT().Get()
+	err = controller.EnsurePlacement(cluster)
+	assert.NoError(t, err)
+
+	placementMock.EXPECT().Get().Return(nil, errors.New("foo"))
+	err = controller.EnsurePlacement(cluster)
+	assert.Error(t, err)
```

Comment:
nit: `assert.Error(t, controller.EnsurePlacement(cluster))`

Reply:
Hmm I worry about breaking convention with the style in the rest of our tests. Also personally I like having the error separate, if you're debugging a test it's easier to do something like `fmt.Printf("%#v", err)` without having to pull it out of the `assert`

Reply:
Okay that's fair 👍